test/studies/diffusion/heat_2d_dist.chpl

```diff
@@ -0,0 +1,51 @@
+use StencilDist;
+
+// setup simulation parameters
+config const xLen : real = 2,
+             yLen : real = 2,
+             nx = 31,
+             ny = 31,
+             nt = 50,
+             sigma = 0.25,
+             nu = 0.05;
+
+const dx : real = xLen / (nx - 1),
+      dy : real = yLen / (ny - 1),
+      dt : real = sigma * dx * dy / nu;
+
+// define a 2D domain and subdomain
+const dom = {0..<nx, 0..<ny},
+      domInner = dom.expand(-1),
+      DOM = dom dmapped Stencil(domInner, fluff=(1, 1)),
+      DOMINNER : subdomain(DOM) = DOM[domInner];
```

Comment:
It seems like a shame to have four domains here when only 2 are used, particularly since `dom` and `domInner` aren't really used after these lines.  I'm also not crazy about the ALLCAPS identifiers.  I was going to suggest just updating the `dom` and `domInner` declarations to minimize the diffs between the two versions (I think it would just be these two lines and the updateFluff), but then I saw the circularity between Stencil wanting to define itself based on domInner...  I'd still probably be inclined to declare this as:

```chapel
const dom = {0..<nx, 0..<ny} dmapped Stencil({1..<nx-1, 1..ny-1}, fluff=(1, 1)),
      domInner = Dom[1..<nx-1, 1..ny-1];
```

A 3-domain, but less repetitive version would be:

```chapel
const locDomInner = {1..<nx-1, 1..<ny-1},
      dom = {0..<nx, 0..<ny} dmapped Stencil(locDomInner, fluff=(1,1)),
      domInner = Dom[locDomInner];
```

Reading this, it's a shame we don't have a 'createArray` factory on Stencil (I don't think anyway), such that we could write:

```chapel
const dom = Stencil.newArray({0..<nx, 0..<ny}, fluff=(1,1)),
      ...
```

(where I'm imagining it would use the fluff values to determine how much of the boundary to discount in computing the block distribution).  Specifically, I think it'd be great to get to a world where we didn't have to lean on `dmapped` so much since it seems mildly unpopular.  Might be worth a feature request issue (or just implementing).